### ISSUE TITLE
Use of GradCache

### ISSUE THREAD

**Issue by aBenNn22** [haon-chen/MoCa#1]
Could you share why did you discontinue using GradCache? Because in my current 8-A100 GPU setup, achieving a batch size of 2048 without GradCache would be technically challenging.

**Comment by haon-chen** [comment]
Hi @aBenNn22 ,
Thank you for your interest in our work! To support training with a larger batch size, we use gradient accumulation (see [this line](https://github.com/haon-chen/MoCa/blob/main/scripts/cl_train.sh#L53) in our training script), which allows us to achieve an effective large batch size without requiring excessive GPU memory. You can increase the **gradient_accumulation_steps** while decreasing the **per_device_train_batch_size**  until the training fits in your resources.
Let us know if you have any further questions!

**Comment by aBenNn22** [comment]
Thank you for answering my question! I will try it. I also have a new question about full fine-tune and lora fine-tune. I'm wondering if the continual pretraining performs worse in lora fine-tune? Or both two stages are worse in lora fine-tune?
